src/main/kotlin/com/healthmetrix/labres/lab/ExtractObxResultUseCase.kt

```diff
@@ -15,23 +16,29 @@ const val OBX_MESSAGE_SEPARATOR = "|"
  */
 @Component
 class ExtractObxResultUseCase {
-    operator fun invoke(message: String, labId: String): LabResult? {
+    operator fun invoke(message: String, labId: String): LabResult? = with(Obx(message)) {
+        result?.let { r -> LabResult(orderNumber, labId, r, testType) }
```

Comment:
can you just do `Obx(message).result?.let { ... }` ? The `with` seems unnecessary

Reply:
No, because I also need the `orderNumber` and `testType` from the Obx